Carbon.IIS/Functions/Set-CIisCollection.ps1

```diff
@@ -0,0 +1,193 @@
+        [String] $LocationPath,
+
+        # The name of the collection to change.
+        [String] $Name,
+
+        # Pass a hashtable for each item that has more than one attribute value to set. Otherwise, pass the attribute
+        # value for the default attribute.
+        [Parameter(ValueFromPipeline)]
+        [Object[]] $InputObject
+    )
+
+    begin
+    {
+        Set-StrictMode -Version 'Latest'
+        Use-CallerPreference -Cmdlet $PSCmdlet -Session $ExecutionContext.SessionState
+
+        function Write-Message
+        {
+            [CmdletBinding()]
+            param(
+                [Parameter(Mandatory, ValueFromPipeline)]
+                [AllowNull()]
+                [AllowEmptyString()]
+                [String] $Message
+            )
+
+            process
+            {
+                if (-not $firstLineWritten)
+                {
+                    Write-Information "Setting IIS configuration collection ${displayPath}."
+                    $firstLineWritten = $true
+                }
+
+                Write-Information "    ${Message}"
+            }
+        }
+
+        $stopProcessing = $false
+        $firstLineWritten = $false
+
+        $getSetArgs = @{}
+        if ($Name)
+        {
+            $getSetArgs['Name'] = $Name
+        }
+
+        $displayPath = ''
+        if ($ConfigurationElement)
+        {
+            $getSetArgs['ConfigurationElement'] = $ConfigurationElement
+            $displayPath = $ConfigurationElement.ElementTagName
+        }
+        else
+        {
+            $getSetArgs['SectionPath'] = $SectionPath
+
+            if ($LocationPath)
+            {
+                $getSetArgs['LocationPath'] = $LocationPath
+            }
+
+            $displayPath =
+                Get-CIisDisplayPath -SectionPath $SectionPath -LocationPath $LocationPath -SubSectionPath $Name
+        }
+
+        $collection = Get-CIisCollection @getSetArgs
+
+        if (-not $collection)
+        {
+            $stopProcessing = $true
+            return
+        }
+
+        $items = [List[hashtable]]::New()
+        $keyValues = @{}
+        $keyAttrName = Get-CIisCollectionKeyName -Collection $collection
+
+        if (-not $keyAttrName)
+        {
+            $msg = "Failed to set IIS configuration collection ${displayPath} because it does not have a key attribute."
+            Write-Error -Message $msg
+            $stopProcessing = $true
+            return
+        }
+    }
+
+    process
+    {
+        if ($stopProcessing)
+        {
+            return
+        }
+
+        foreach ($item in $InputObject)
+        {
+            if ($item -isnot [IDictionary])
+            {
+                $item = @{ $keyAttrName = $item }
+            }
+
+            $items.Add($item)
+            $keyValues[$item[$keyAttrName]] = $true
+        }
+    }
+
+    end
+    {
+        if ($stopProcessing)
+        {
+            return
+        }
+
+        $itemsToRemove =

```

Comment:
Write a message indicating what was removed. Unless the other functions already do that?

Reply:
The other function does that.